File: client/src/app/models/Product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  category: string;
  brand: string;
  tags: string[];
  price: number;
  discountPercentage: number;
  rating: number;
  warrantyInformation: string;
  shippingInformation: string;
  availabilityStatus: string;
  quantityInStock: number;
  returnPolicy: string;
  images: string[];
  thumbnail: string;
}

export interface ProductParams {
  orderBy: string;
  searchTerm?: string;
  brands?: string[];
  categories?: string[];
  pageSize: number;
  pageNumber: number;
}

File: server/Entity/Product.cs
namespace server.Entity
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string Brand { get; set; }
        public List<string> Tags { get; set; }
        public long Price { get; set; }
        public double DiscountPercentage { get; set; }
        public double Rating { get; set; }
        public string WarrantyInformation { get; set; }
        public string ShippingInformation { get; set; }
        public string AvailabilityStatus { get; set; }
        public int QuantityInStock { get; set; }
        public string ReturnPolicy { get; set; }
        public List<string> Images { get; set; }
        public string Thumbnail { get; set; }
    }
}